backend/src/tests.rs

```diff
@@ -136,18 +170,77 @@ mod test {
         Ok(())
     }
 
+    #[tokio::test]
+    async fn test_concurrent_proof_submissions() -> Result<(), Box<dyn Error>> {
```

Comment:
Can you make much clear in the commments what is the goal of this test and how the results you are obtaining are proving that the goal was accomplished?